packages/backend/src/nest/registration/registration.service.ts

```diff
@@ -75,10 +99,11 @@ export class RegistrationService extends EventEmitter implements OnModuleInit {
   }
 
   public async registerUserCertificate(csr: string): Promise<void> {
-    const result = await issueCertificate(csr, this._permsData)
+    const result = await issueCertificate(csr, this.permsData)
     this.logger('DuplicatedCertBug', { result })
     if (result?.cert) {
-      this.emit(RegistrationEvents.NEW_USER, { certificate: result.cert })
+      // @ts-ignore
+      await this.storageService?.saveCertificate({ certificate: result.cert })
```

Comment:
We only process the next certificate after we have saved the current batch of certificates and we know we have saved them because we are awaiting it here rather than sending events which are not awaited.